src/ccontrol/UserQuerySelect.cc

```diff
@@ -233,6 +233,11 @@ std::string UserQuerySelect::getResultQuery() const {
 
 /// Begin running on all chunks added so far.
 void UserQuerySelect::submit() {
+    auto exec = _executive;
+    if (exec == nullptr) {
+        LOGS(_log, LOG_LVL_ERROR, "UserQuerySelect::submit() executive is null at start");
```

Comment:
This is a bit confusing. Shouldn't the user query object be constructed with a valid pointer to the "executive"? And if so then why is this test? What makes the pointer to suddenly disappear?
I'm also seeing the very same pattern of checking for the validity of a pointer all around the code of the PR. Perhaps there is a safer way of constructing objects to avoid putting the code with these tests?
My other worry is that if there is such an uncertainty (with the elusive pointer) then such code must be really hard to debug.

Reply:
_executive being null at this point should really never happen, so that can be removed.